packages/react-slider/src/components/Slider/useSliderState.tsx

```diff
@@ -0,0 +1,236 @@
+    onKeyDown: onKeyDownCallback,
+  } = state;
+
+  const [currentValue, setCurrentValue] = useControllableValue(
+    value && clamp(value, min, max),
+    clamp(defaultValue, min, max),
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any -- ev is not a SyntheticEvent
+    (ev: any, val) => onChange?.(val!, ev),
+  );
+
+  const railRef = React.useRef<HTMLDivElement>(null);
+  const thumbRef = React.useRef<HTMLDivElement>(null);
+  const disposables = React.useRef<(() => void)[]>([]);
+  const id = useId('Slider', state.id);
+
+  /**
+   * Updates the `currentValue` to the new `incomingValue` and clamps it.
+   *
+   * @param ev
+   * @param incomingValue
+   */
+  const updateValue = React.useCallback(
+    (ev, incomingValue: number): void => {
+      if (incomingValue !== min && incomingValue !== max) {
+        ev.stopPropagation();
+        if (ev.type === 'keydown') {
+          ev.preventDefault();
+        }
+      }
+
+      setCurrentValue(clamp(incomingValue, min, max), ev);
+    },
+    [max, min, setCurrentValue],
+  );
+
+  /**
+   * Calculates the `step` position based off of a `Mouse` or `Touch` event.
+   */
+  const calculateSteps = React.useCallback(
+    (ev): number => {
+      const currentBounds = railRef?.current?.getBoundingClientRect();
+      const size = currentBounds?.width || 0;
+      const position = currentBounds?.left || 0;
+
+      const totalSteps = (max - min) / step;
+      const stepLength = size / totalSteps;
+      const thumbPosition = ev.clientX;
+      const distance = thumbPosition - position;
+      return distance / stepLength;
+    },
+    [max, min, step],
+  );
+
+  const onPointerMove = React.useCallback(
+    (ev): void => {
+      if (snapToStep || step !== 1) {
+        updateValue(ev, Math.round((min + step * calculateSteps(ev)) / step) * step);
+      } else {
+        updateValue(ev, min + step * calculateSteps(ev));
+      }
+    },
+    [calculateSteps, min, snapToStep, step, updateValue],
+  );
+
+  const onPointerUp = (): void => {
+    disposables.current.forEach(dispose => dispose());
+    disposables.current = [];
+  };
+
+  const onPointerDown = React.useCallback(
+    (ev): void => {
+      const { currentTarget, pointerId } = ev;
+
+      onPointerDownCallback?.(ev);
+
+      if (currentTarget.setPointerCapture) {
+        currentTarget.setPointerCapture(pointerId);
+      }
+
+      disposables.current.push(
+        on(currentTarget, 'pointermove', onPointerMove),
+        on(currentTarget, 'pointerup', onPointerUp),
+        () => {
+          currentTarget.releasePointerCapture(pointerId);
+        },
+      );
+
+      onPointerMove(ev);
+    },
+    [onPointerMove, onPointerDownCallback],
+  );
+
+  const onKeyDown = React.useCallback(
+    (ev: React.KeyboardEvent<HTMLDivElement>): void => {
+      const key = getCode(ev);
+
+      onKeyDownCallback?.(ev);
+
+      if (ev.shiftKey) {
+        if (key === ArrowDownKey || key === ArrowLeftKey) {
```

Comment:
`ArrowLeftKey` should flip in RTL to `ArrowRightKey`. Am I missing something?

Reply:
They should flip in RTL. I’m planning on adding that and the Slider movement modification in a separate pull request to lower the scope here.